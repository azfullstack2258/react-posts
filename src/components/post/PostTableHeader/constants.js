const COLUMNS = [
  {
    key: 'id',
    label: 'ID'
  },
  {
    key: 'property',
    label: 'Property'
  },
  {
    key: 'user',
    label: 'User'
  },
  {
    key: 'network',
    label: 'Network'
  },
  {
    key: 'date',
    label: 'Date'
  },
  {
    key: 'image',
    label: 'Image'
  },
  {
    key: 'content',
    label: 'Content'
  }
];

export { COLUMNS };
