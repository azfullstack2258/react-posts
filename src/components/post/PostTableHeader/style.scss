.post-table__header {
  display: flex;
  background: #343a40;

  .post-table__row-cell {
    white-space: nowrap;
    min-width: fit-content;
    color: white;
    font-weight: bold;
    padding: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 30px;

    &:not(.content) {
      border-right: 1px solid #464d55;
    }

    &.id {
      width: 30px;
    }

    &.user {
      min-width: 120px;
      max-width: 120px;
    }

    &.property,
    &.network,
    &.date,
    &.image {
      min-width: 100px;
      max-width: 100px;
    }

    &.content {
      flex-grow: 1;
    }
  }
}