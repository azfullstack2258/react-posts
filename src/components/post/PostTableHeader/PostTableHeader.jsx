import React from 'react';

import { COLUMNS } from './constants';
import './style.scss';

const PostTableHeader = () => (
  <div className="post-table__header">
    {COLUMNS.map(({ key, label }) => (
      <div key={key} className={`post-table__row-cell ${key}`}>
        {label}
      </div>
    ))}
  </div>
);

export default PostTableHeader;
