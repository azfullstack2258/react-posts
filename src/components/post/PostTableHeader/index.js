import PostTableHeader from './PostTableHeader';

export default PostTableHeader;
