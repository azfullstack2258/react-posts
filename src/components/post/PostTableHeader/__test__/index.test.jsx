import React from 'react';
import { render } from '@testing-library/react';

import PostTableHeader from '../PostTableHeader';

describe('<PostTableHeader />', () => {
  it('should render a prop', () => {
    const { container } = render(
      <PostTableHeader />
    );

    expect(container).toMatchSnapshot();
  })
});
