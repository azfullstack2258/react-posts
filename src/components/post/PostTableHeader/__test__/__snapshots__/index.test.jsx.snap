// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTableHeader /> should render a prop 1`] = `
<div>
  <div
    class="post-table__header"
  >
    <div
      class="post-table__row-cell id"
    >
      ID
    </div>
    <div
      class="post-table__row-cell property"
    >
      Property
    </div>
    <div
      class="post-table__row-cell user"
    >
      User
    </div>
    <div
      class="post-table__row-cell network"
    >
      Network
    </div>
    <div
      class="post-table__row-cell date"
    >
      Date
    </div>
    <div
      class="post-table__row-cell image"
    >
      Image
    </div>
    <div
      class="post-table__row-cell content"
    >
      Content
    </div>
  </div>
</div>
`;
